### PR TITLE
Updates on issue1237

### PR DESCRIPTION
**Pull request by YufeiCui**
Before submitting a final PR, please:

- [x] Run `npm test` on your machine
- [x] Run `npm run format` to keep code formatted consistently

### CODE REVIEW

File: extension/history.js [hunk 1]
```diff
@@ -20,107 +20,110 @@ export class Database {
...
 
   getAll(TBName) {
-    return new Promise(resolve => {
-      const request = indexedDB.open(this.DBName);
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const objectStore = database.transaction(TBName).objectStore(TBName);
-        const list = [];
```
Comment (YufeiCui) [inline extension/history.js:58]: @OliviaRuyinZhang Changed this to use getAll(), according to this

image

Let me know if you have comments on this, i.e., if you prefer the way you did it.

**Review by paulZhongym — APPROVED** [review]

File: extension/history.js [hunk 2]
```diff
@@ -20,107 +21,124 @@ export class Database {
...
           .transaction([TBName], this.readwrite)
           .objectStore(TBName)
           .delete(primaryKey);
-        remove.onerror = event =>
-          reject("This entry could not be removed from database.");
-        remove.onsuccess = event =>
-          resolve("This entry has been removed from your database.");
+        remove.onsuccess = e => {
+          log.info("1 record has been added to your database.");
+          resolve();
```
Comment (michael-mml) [inline extension/history.js:124]: I believe this should log `This entry has been removed from your database`.

Comment (YufeiCui) [reply extension/history.js:124]: Fixed!

File: extension/history.js [hunk 3]
```diff
@@ -20,107 +21,124 @@ export class Database {
...
         const add = database
           .transaction([TBName], this.readwrite)
           .objectStore(TBName)
           .add(obj);
-        add.onsuccess = event => {
-          // log.info("1 record has been added to your database.");
-          resolve("1 record has been added to your database.");
+        add.onsuccess = e => {
+          log.info("1 record has been added to your database.");
+          resolve();
```
Comment (michael-mml) [inline extension/history.js:102]: Also here, should the user get back the entry they added? For example, if IndexedDB adds an `id` field, whoever calls add will not have `id` available to them.

File: extension/background/intentRunner.js [hunk 4]
```diff
@@ -360,7 +361,7 @@ export function getIntentSummary() {
 function addIntentHistory(context) {
   intentHistory.push(context);
   intentHistory.splice(0, intentHistory.length - INTENT_HISTORY_LIMIT);
-  db.add(context, utteranceTable);
+  db.add(context, utteranceTable).catch(error => log.error(error));
```
Comment (michael-mml) [inline extension/background/intentRunner.js]: This might be a good place to use `async and await`.

File: extension/history.js [hunk 5]
```diff
@@ -20,107 +20,131 @@ export class Database {
...
-      request.onerror = function(e) {
-        reject("Error: failed to create a database!");
+      request.onerror = e => {
+        const code = e.target.errorCode;
+        reject(new Error(`Failed to create a database: ${code}`));
       };
     });
   }
 
   get(primaryKey, TBName) {
```
Comment (michael-mml) [inline extension/history.js]: TBName should probably be changed to tbName if we're following other conventions

**Review by michael-mml — APPROVED** [review]
Good.

**Comment by michael-mml** [comment]
@atafliovich Hi Professor Anya, this issue was merged as linked [here](https://github.com/mozilla/firefox-voice/pull/1303).

For deliverable 2, we merged this [issue](https://github.com/mozilla/firefox-voice/pull/1152) (since as I was mainly working on this, I forked it into my own repository and we also did not have the team repository at this time).
We also merged this [issue](https://github.com/mozilla/firefox-voice/pull/1227) for deliverable 2.

**Pull request merged**